fdbserver/DDRelocationQueue.actor.cpp

```diff
@@ -530,6 +530,35 @@ struct DDQueue : public IDDRelocationQueue {
 		Future<Void> cancel;
 	};
 
+	struct MovingAverage {
+		int64_t previous;
+		int64_t total;
+		double interval;
+		Deque<std::pair<double, int64_t>> updates; // std::pair<time, bytes>
+
+		int64_t getTotal() { return total; }
+
+		double getAverage() {
+			while (!updates.empty() && updates.front().first < now() - interval) {
+				previous += updates.front().second;
+				updates.pop_front();
+			}
+			return (total - previous) / interval;
+		}
+
+		void sumBytes(int64_t bytes) {
+			total += bytes;
+			updates.push_back(std::make_pair(now(), bytes));
+		}
+
+		MovingAverage() {
```

Comment:
Is the initialization function needed? Can the members be intialized by default where they're declared above?